File: CSharp/ByTier/A/1926_A_Vlad_and_the_Best_of_Five.cs
// Problem: CF 1926 A - Vlad and the Best of Five
// https://codeforces.com/contest/1926/problem/A

using System;
class Program {
  static void Main(string[] args) {
    int t = int.Parse(Console.ReadLine()); // Read number of test cases
    for (int i = 0; i < t; i++) {
      Solve(); // Process each test case
    }
  }
  static void Solve() {
    string s = Console.ReadLine().Trim(); // Read the input string
    int countA = 0, countB = 0; // Initialize counters for A and B
    foreach (char c in s) { // Iterate through each character in the string
      if (c == 'A') {
        countA++; // Increment A counter if character is A
      } else {
        countB++; // Increment B counter if character is B
      }
    }
    Console.WriteLine(countA > countB ? "A" : "B"); // Print the character with higher count
  }
}
/*
Purpose: This program determines which character ('A' or 'B') appears most frequently in a 5-character string.

Algorithms/Techniques: 
- Single pass iteration through string to count occurrences
- Simple comparison to determine maximum

Time Complexity: O(n) where n is the length of the string (5 in this case)
Space Complexity: O(1) - only using a constant amount of extra space
*/

// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/A/1926_A_Vlad_and_the_Best_of_Five.py
# Problem: CF 1926 A - Vlad and the Best of Five
# https://codeforces.com/contest/1926/problem/A

"""
Problem: A. Vlad and the Best of Five
Algorithm: Counting characters in a string
Time Complexity: O(n) where n is the length of the string (5 in this case)
Space Complexity: O(1) - only using a constant amount of extra space

The solution counts occurrences of 'A' and 'B' in each string and prints the character with higher count.
"""

testcases = int(input())
for _ in range(testcases):
    s = input()
    count_a = 0
    count_b = 0
    for i in range(len(s)):
        if s[i] == "A":
            count_a += 1
        elif s[i] == "B":
            count_b += 1
    if count_a > count_b:
        print("A")
    else:
        print("B")


# https://github.com/VaHiX/CodeForces/